### ISSUE TITLE
Keyword Argument Filtering for Metric Compositions/Collections does not behave as expected

### ISSUE THREAD

**Issue by Chrixtar** [PyTorchLightning/metrics#693]
## 🐛 Bug
Metric collections / compositions (using metric arithmetics) do not filter keyword arguments based on update signature, if a metric does not expect any of the given keyword arguments.

### To Reproduce
```python
import torch
from torchmetrics import MetricCollection, Metric, Accuracy


class DummyMetric(Metric):
    def __init__(self):
        super(DummyMetric, self).__init__()

    def update(self, *args, kwarg):
        print("Entered DummyMetric")

    def compute(self):
        return


class MyAccuracy(Metric):
    def __init__(self):
        super(MyAccuracy, self).__init__()

    def update(self, preds, target, kwarg2):
        print("Entered MyAccuracy")

    def compute(self):
        return


if __name__ == "__main__":
    mc = MetricCollection([Accuracy(), DummyMetric()])      # Accuracy() + MyOwnMetric()
    mc2 = MetricCollection([MyAccuracy(), DummyMetric()])
    try:
        mc(torch.tensor([0, 1]), torch.tensor([0, 1]), kwarg="kwarg")
    except TypeError as e:
        print(e)
    mc2(torch.tensor([0, 1]), torch.tensor([0, 1]), kwarg="kwarg", kwarg2="kwarg2")
```
The output of this function is as follows:
```console
update() got an unexpected keyword argument 'kwarg'
Entered MyAccuracy
Entered MyAccuracy
Entered DummyMetric
Entered DummyMetric
```
### Expected Behavior
If a metric does not expect any of the given keyword arguments, it should be called with positional arguments only.

**Comment by Borda** [comment]
@justusschock mind have a look :rabbit:

**Comment by justusschock** [comment]
@Borda the issue is this line: https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/metric.py#L566 if no kwarg was found, the filter mecahanism returns all kwargs to match possible **kwargs. We could check if some **kwargs exist and only in that case return all kwargs
